src/v/kafka/server/handlers/fetch.cc

```diff
@@ -672,6 +672,501 @@ class parallel_fetch_plan_executor final : public fetch_plan_executor::impl {
     }
 };
 
+/**
+ * A fetch exeutor that relies on notifications from raft to determine when to
+ * query partitions. Rather than querying all partitions at a set polling
+ * interval like the `parallel_fetch_plan_executor`.
+ */
+class nonpolling_fetch_plan_executor final : public fetch_plan_executor::impl {
+    // Passed from the coordinator shard to fetch workers.
+    // Contains either references to objects local to the fetch worker shard or
+    // copies of data from the coordinator shard.
+    struct shard_local_fetch_context {
+        bool foreign_read;
+        size_t bytes_left;
+        size_t min_bytes;
+        std::optional<model::timeout_clock::time_point> deadline;
+        std::vector<ntp_fetch_config> requests;
+        server& srv;
+        cluster::partition_manager& mgr;
+        ss::abort_source& as;
+    };
+
+    // Returned by fetch workers to the coordinator shard.
+    struct worker_result {
```

Comment:
please comment the members of this struct

Reply:
Adding comments